### PR TITLE
"Star Rewards" dialog

### PR DESCRIPTION
**Pull request by balderdash**
We don't actually use stars in this dialog, but "points":

image

The point total is stored in localstorage as a map from level id to points scored on that level.

### CODE REVIEW

**Comment by joshlory** [comment]
Worth implementing the "3 points" badge with CSS `text-shadow:` and `transform: rotate` (or an SVG) instead of the static images?

File: apps/i18n/common/en_us.json [hunk 1]
```diff
@@ -225,6 +225,7 @@
   "more": "More",
   "nestedForSameVariable": "You're using the same variable inside two or more nested loops. Use unique variable names to avoid infinite loops.",
   "nextLevel": "Congratulations! You completed Puzzle {puzzleNumber}.",
+  "nextLevelStars": "You completed Puzzle {puzzleNumber}! You have earned a total of {numPoints} points.",
```
Comment (joshlory) [inline apps/i18n/common/en_us.json]: Nit: `{numPoints, plural, one {1 point} other {# points}}.`

File: apps/src/templates/AchievementDialog.jsx [hunk 2]
```diff
@@ -157,8 +229,19 @@ const AchievementDialog = Radium(React.createClass({
...
+    let feedbackMessage;
+    if (this.props.bannerMode) {
+      feedbackMessage = locale[tooManyBlocks ? 'numBlocksNeeded' : 'nextLevelStars'](params);
+    } else {
+      feedbackMessage = locale[tooManyBlocks ? 'numBlocksNeeded' : 'nextLevel'](params);
+    }
+    const numPoints = 1 +
+      (showNumBlocksRow && !tooManyBlocks ? 1 : 0) +
+      (!tooManyHints ? 1 : 0);
+    const badgeUrl = `url(${this.props.assetUrl(`media/dialog/badges/${numPoints}-points.png`)})`;
```
Comment (joshlory) [inline apps/src/templates/AchievementDialog.jsx]: Nit: assumes `numPoints <= 10` 😛

File: apps/src/templates/AchievementDialog.jsx [hunk 3]
```diff
@@ -207,6 +298,18 @@ const AchievementDialog = Radium(React.createClass({
...
                 }
+                {this.props.bannerMode &&
+                  <div
+                    style={{
+                      ...styles.bannerAchievement.banner,
+                      backgroundImage: dotsUrl,
+                    }}
+                  >
+                    <div style={styles.bannerAchievement.bannerText}>
+                      CONGRATULATIONS
```
Comment (joshlory) [inline apps/src/templates/AchievementDialog.jsx]: I18n?

**Comment by balderdash** [comment]
Here's what I managed to get with just html/css on a blank badge:
image

**Pull request merged**